app/helpers/common_parameters_helper.rb

```diff
@@ -16,17 +16,38 @@ def parameter_value_field(value)
   end
 
   def parameter_value_content(id, value, options)
-    content_tag(:span, options[:popover], :class => "input-group-addon") +
-      text_area_tag(id, value, { :rows => 1,
-                                 :class => 'form-control no-stretch',
-                                 :'data-property' => 'value',
-                                 :'data-hidden-value' => Parameter.hidden_value,
-                                 :'data-inherited-value' => options[:inherited_value],
-                                 :name => options[:name].to_s,
-                                 :disabled => options[:disabled] })
+    lookup_key = options[:lookup_key]
+
+    option_hash = { :rows => 1,
+                    :class => 'form-control no-stretch',
+                    :'data-property' => 'value',
+                    :'data-hidden-value' => LookupKey.hidden_value,
+                    :'data-inherited-value' => options[:inherited_value],
+                    :name => options[:name].to_s,
+                    :disabled => options[:disabled] }
+
+    if lookup_key.present? && lookup_key.hidden_value?
+      field = password_field_tag(id, value, option_hash)
+    else
+      field = text_area_tag(id, value, option_hash)
+    end
+
+    content_tag(:span, options[:popover], :class => "input-group-addon") + field
   end
 
   def use_puppet_default_help link_title = nil, title = _("Use Puppet default")
     popover(link_title, _("Do not send this parameter via the ENC.<br>Puppet will use the value defined in the manifest."), :title => title)
   end
+
+  def hidden_value_field(f, field, value, disabled, options = {})
+    if f.object.hidden_value?
+      field = f.password_field(field, :disabled => disabled, :value => value, :class => 'form-control')
+    else
+      field = f.text_area(field, options.merge(:disabled => disabled,
```

Comment:
In changing the name here it's reassigning/overlapping with the existing `field` variable, I'd avoid doing that to keep the code clearer.